src/org/labkey/test/components/domain/BaseDomainDesigner.java

```diff
@@ -0,0 +1,100 @@
+package org.labkey.test.components.domain;
+
+import org.labkey.test.BootstrapLocators;
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Automates the LabKey ui component defined in: packages/components/src/components/domainproperties/BaseDomainDesigner.tsx
+ * Defines standard save, cancel, and error handling
+ */
+public abstract class BaseDomainDesigner<EC extends BaseDomainDesigner.ElementCache> extends WebDriverComponent<EC>
+{
+    private final WebElement el;
+    private final WebDriver driver;
+
+    protected BaseDomainDesigner(WebDriver driver)
+    {
+        this.driver = driver;
+        el = Locator.id("app").findElement(this.driver); // Full page component
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return driver;
+    }
+
+    public boolean isCancelButtonEnabled()
+    {
+        return elementCache().cancelButton.isEnabled();
+    }
+
+    public Object clickCancel()
+    {
+        clickCancel(true);
+        return null;
+    }
+
+    protected final void clickCancel(boolean expectPageLoad)
+    {
+        if (expectPageLoad)
+        {
+            getWrapper().doAndWaitForPageToLoad(() -> elementCache().cancelButton.click());
+        }
+        else
+        {
+            elementCache().cancelButton.click();
+        }
+    }
+
+    public boolean isSaveButtonEnabled()
+    {
+        return elementCache().saveButton.isEnabled();
+    }
+
+    public Object clickSave()
+    {
+        clickSave(true);
+        return null;
+    }
+
+    protected final void clickSave(boolean expectPageLoad)
+    {
+        if (expectPageLoad)
+        {
+            getWrapper().doAndWaitForPageToLoad(() -> elementCache().saveButton.click());
+        }
+        else
+        {
+            elementCache().saveButton.click();
+        }
+    }
+
+    public List<String> clickSaveExpectingErrors()
+    {
+        elementCache().saveButton.click();
+        return getWrapper().getTexts(BootstrapLocators.errorBanner.waitForElements(getWrapper().shortWait()));
```

Comment:
My experience with the .getTexts call is that if there is any kind of a mark-up in the message it can add spaces at the front and end of the text. This becomes a tricky for text expecting specific text in the error message.
Maybe this code should do some clean up of the text before adding them to the list.

Reply:
I wasn't aware of that. I've updated it to `.trim()` the error texts.